### ISSUE TITLE
FR: Countdown - Contact form (2 of 4)

### ISSUE THREAD

**Issue by Meli-lazarte** [aemsites/vg-macktrucks-com#145]
### Description
As a long-haul customer, I want to provide my email address to Mack Trucks so that I can stay informed about the launch of the new MT.
 
### Context
A contact form should be created for the countdown component created in #133. 

### AC

- [x] The contact form has to be added with an iframe to the v2-hero countdown variant.
- [ ] A confirmation email is sent to the user (testing only)
- [x] Mobile and desktop testing is done.
- [ ] This will use a MCAE (Marketing Cloud Account Engagement) form using iframe solution (not new form handler block request): iFrame URL - https://ma.macktrucks.com/l/1038343/2025-01-09/3mttsb

### Out of scope

- ~Checkbox functionality will be handled in a separate ticket.~
- A confirmation message is displayed when submitting the form: "You're locked in for future updates." - this will be handled as part of the iframe and has been configured in MCAE. 
- Modification of styles are out of scope.
- Testing of email confirmation

### Figma
https://www.figma.com/design/apGFsQv4eiZuJ9N6DxCFIm/M1-Teaser-landing-page?node-id=0-1&t=tmh2YYAhwPJpN9lX-1

### Notes

- ~~The V2 forms block will be used~~
- This functionality should be available for all the market although LM is out of scope for M1.
- ~~The block being updated is the hero block.~~
- For mobile view the checkbox is the same as the one in desktop.
- The values for the checkbox is "true" or "false".
- Here are the field names: 
email
marketignConsent
![image](https://github.com/user-attachments/assets/5ef2aa84-e774-41fb-bab4-874b507fc62f)

**Comment by Meli-lazarte** [comment]
We still need to know what data will be requested in the form. Conversation with Ryan.

**Comment by BeckyMedlin** [comment]
Waiting on final designs per @ZoeBeykirch and @WendyKruger

**Comment by ZoeBeykirch** [comment]
Designs (added to ticket), we need to add in AC for this next - https://www.figma.com/design/apGFsQv4eiZuJ9N6DxCFIm/M1-Teaser-landing-page?node-id=2-449&t=sXFz6XFijxKwb1RN-1

**Comment by Meli-lazarte** [comment]
Design for the pop-up confirmation message should be added to Figma (we have a pic for now).

**Comment by ryanropero** [comment]
Here is the link to the form handler: 
https://ma.macktrucks.com/l/1038343/2025-01-14/3mtzyy

**Comment by ryanropero** [comment]
Here are the field names: 
email
marketignConsent
![image](https://github.com/user-attachments/assets/5ef2aa84-e774-41fb-bab4-874b507fc62f)

**Comment by Meli-lazarte** [comment]
Create a separate ticket for the checkbox functionality.

**Comment by ZoeBeykirch** [comment]
@alan-le-bihan I have updated the ticket to remove the popup message from acceptance criteria, this will be handled in the iFrame and has been configured in MCAE so no dev work will actually be needed. Thanks!

**Comment by WendyKruger** [comment]
@alan-le-bihan here is the url for the form to use in the iframe https://ma.macktrucks.com/l/1038343/2025-01-09/3mttsb

**Comment by VladMano** [comment]
The following issues have been found after initial testing:

Desktop:

- A confirmation e-mail is not being sent out after filling up the form.
- After filling up the form and pressing on “Stay informed” the “You’re locked in for future updates.” Text appears, the text below it is not displayed correctly: 
![Image](https://github.com/user-attachments/assets/a8746bd6-449c-48a0-bc40-c4a4228d225c)
After filling up the form and pressing on the “Watch Video” button, the video seems to be broken:
![Image](https://github.com/user-attachments/assets/1410bc09-de4e-4be4-9a24-5ccd38d4fafa)

Mobile:
- The bottom part of the contact form is being cut off as the whole form does not fit on screen (This has been tested on an actual device, S24 Ultra and on browserstack emulation with iPhone 15 Pro Max):
![Image](https://github.com/user-attachments/assets/b3801fc9-65c8-49a4-9411-07528c840b25)
https://github.com/user-attachments/assets/71973df9-f9ad-4ff7-b771-eaff8b0b5488

**Comment by alan-le-bihan** [comment]
@VladMano The form is embedded using an iframe, so all of its functionality isn't managed on our end.

**Comment by BeckyMedlin** [comment]
@alan-le-bihan -- @WendyKruger is working on the styling today. 

Confirmed with Casi McDaniel that they do not have the content for an autoresponder email yet. 

Form will be removed on April 3 and will need to display only the countdown.

**Comment by BeckyMedlin** [comment]
@alan-le-bihan Let's plan to release what you've been able to create this sprint. If styling is incomplete, @WendyKruger will need to modify before it will be ready for publication. Please unblock so we can test what you've done. Thanks!

**Comment by alan-le-bihan** [comment]
@BeckyMedlin What I did has already been tested by @VladMano, so I'll update the ticket status.

**Comment by BeckyMedlin** [comment]
Closing as work has been deployed.
